src/python/pants/backend/python/macros/poetry_requirements_test.py

```diff
@@ -0,0 +1,414 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from textwrap import dedent
+from typing import Iterable
+
+import pytest
+from _pytest.logging import LogCaptureFixture
```

Comment:
This is a private import, so typically best to not import it because there's less guarantee if it will break overnight when changing Pytest versions.

This does mean that `caplog` won't have a type hint, which is okay in this case because it's a test and `caplog` is well understood by most readers what it is. (Ack that you had spent some time trying to find the type hint)